### DISCUSSION TITLE
Schrödinger's Multilevel Switch CC

### DISCUSSION THREAD

**Discussion by AlCalzone** [zwave-js/zwave-js#1459] [General] [answered]
@hanskroner [This](https://github.com/zwave-js/node-zwave-js/issues/1426) is interesting. We have a device that supports `Multilevel Switch CC` and responds to non-encapsulated commands:
```
21:17:11.475 DRIVER » [Node 022] [REQ] [SendData]
                      │ transmit options: 0x25
                      │ callback id:      11
                      └─[MultilevelSwitchCCGet]
...
21:17:11.528 DRIVER « [Node 022] [REQ] [ApplicationCommand]
                      └─[MultilevelSwitchCCReport]
                          current value: 99
                          target value:  99
                          duration:      [Duration: 0seconds]
```

It supports supervision, but does not support `Multilevel Switch Set` in `Supervision`:
```
21:22:22.791 DRIVER » [Node 022] [REQ] [SendData]
                      │ transmit options: 0x25
                      │ callback id:      88
                      └─[SupervisionCCGet]
                        │ session id:      2
                        │ request updates: true
                        └─[MultilevelSwitchCCSet]
                            target value: 0
...
21:22:22.837 DRIVER « [Node 022] [REQ] [ApplicationCommand]
                      └─[SupervisionCCReport]
                          session id:          2
                          more updates follow: false
                          status:              NoSupport
                          duration:            [Duration: 0seconds]
```
Is there a way to determine which commands we can send supervised and which we cannot? Other than to try it out and check the `status` flag?

I ran into a similar issue where the Set command was supported with encapsulation, but the `Start/Stop Level Change` were not. For these two commands I remember that supervision isn't supported and send them non-encapsulated.

**Comment by hanskroner** [comment, marked as answer]
> Is there a way to determine which commands we can send supervised and which we cannot? Other than to try it out and check the `status` flag?

Unfortunately, no. Or at least, not that I'm aware of. It'd be an interesting question to bring up in the Application Working Group and hear if some mechanism for querying which commands are supported has been considered.

Without that, trying Supervision and keeping a "blacklist" of which commands aren't supported sounds reasonable, if a bit inconvenient 😐
